implementations/rust/ockam/ockam_transport_tcp/tests/send_receive.rs

```diff
@@ -0,0 +1,65 @@
+use core::iter;
+use std::time::Duration;
+
+use ockam_core::compat::sync::Arc;
+use ockam_core::{route, Error, Result, Routed, Worker};
+use ockam_node::Context;
+use rand::Rng;
+use tokio::sync::{oneshot, RwLock};
+use tokio::time::timeout;
+
+use ockam_transport_tcp::{TcpTransport, TCP};
+
+#[tokio::test(flavor = "multi_thread")]
+async fn send_receive() {
+    let (tx, rx) = oneshot::channel::<Result<()>>();
+    let (ctx, mut executor) = ockam_node::start_node();
+    executor
+        .execute(async move { run_with_timeout(tx, ctx).await })
+        .expect("Executor should not fail");
+    assert!(rx.await.unwrap().is_ok(), "'run' method should return Ok");
+}
+
+async fn run_with_timeout(tx: oneshot::Sender<Result<()>>, ctx: Context) {
+    let ctx = Arc::new(RwLock::new(ctx));
```

Comment:
Maybe we can use ctx.new_context() which will spawn new instance of Context with new address, therefore Arc and RwLock won't be needed

Reply:
I've done a small refactor to avoid the use of Arc and RwLock.